cognite/neat/rules/importers/_spreadsheet2rules.py

```diff
@@ -131,25 +131,24 @@ def sheet_names(self, role: RoleTypes) -> set[str]:
         names = MANDATORY_SHEETS_BY_ROLE[role]
         return {f"{self._sheet_prefix}{sheet_name}" for sheet_name in names if sheet_name != "Metadata"}
 
-    def read(self, filepath: Path) -> None | ReadResult:
-        with pd.ExcelFile(filepath) as excel_file:
-            self._seen_files.add(filepath)
-            self._seen_sheets.update(map(str, excel_file.sheet_names))
-            metadata: MetadataRaw | None
-            if self.metadata is not None:
-                metadata = self.metadata
-            else:
-                metadata = self._read_metadata(excel_file, filepath)
-                if metadata is None:
-                    # The reading of metadata failed, so we can't continue
-                    return None
-
-            sheets, read_info_by_sheet = self._read_sheets(excel_file, metadata.role)
-            if sheets is None or self.issue_list.has_errors:
+    def read(self, excel_file: pd.ExcelFile, filepath: Path) -> None | ReadResult:
```

Comment:
Switch this one to opening the excel file outside. It is faster than opening the excel files for each set of sheets we want to read.